robottelo/containers.py

```diff
@@ -0,0 +1,247 @@
+import random
+from uuid import uuid4
+from robottelo.config import settings
+
+
+class ContainerError(Exception):
+    """Exception raised for failed container management operations."""
+
+
+class DockerContainer:
+    """
+    Gather information needed to spin up a Docker-based container.
+    Attributes:
+            image (init): The name of the container image to build from.
+            tag (init): The tag of the operating system to use.
+            agent (boolian): If set to True, prepares container for
+            katello-agent.
+    """
+
+    def __init__(self, image, tag, agent, ports):
+        """
+        Parameters:
+            image (init): The name of the container image to build from.
+            tag (init): The tag of the operating system to use.
+            agent (boolian): If set to True, prepares container for
+            katello-agent.
+        Returns:
+            A container object if tag and image supplied.
+        """
+        import docker
+        import docker.api.container
+
+        self._client = docker.Client(version="1.22")
+        self.image = image
+        self.tag = tag
+        self.name = uuid4()
+        self.ports = ports or {}
+        self._inst = None
+        self._logs = {}
+        self._mount = agent
+        self._create()
+
+    def _create(self):
+        """Creates containers for an instance of this class.
+        Mounts /dev/log as a read-write file system if katello-agent
+        is required.
+        Prints RHSM logs to stout.
+        Prints host name to stout after starting the container.
+        """
+        print(f"Creating {self.tag} container named {self.name}")
```

Comment:
Ahh @latran, we'll need to switch all print statements to logging statements

Reply:
Made the necessary change but do double check on my work.